api-get-object.go

```diff
@@ -289,6 +292,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, syscall.EBADF
```

Comment:
since we always return type `ErrorResponse` for error it's better to:
`return ErrInvalidObjectName("ETag has changed")`

instead of returning syscall.EBADF

But this sounds better:
`return ErrInvalidObject("ETag has changed")`

we could rename `ErrInvalidObjectName()` to `ErrInvalidObject()`